_posts/2022-01-21-dashboards-plugins-intro.markdown

```diff
@@ -0,0 +1,220 @@
+
+![DashboardsOverview](/assets/media/blog-images/2022-01-21-dashboards-plugin-intro/dashboards_overview.png){:width="100%"}
+
+It has 3 main components:
+
+1. Core: The main runtime responsible for managing the lifecycle of the application and all its main services
+2. Packages: A set of static utilities that can be imported and used throughout the application (Both by core and plugins).
+3. Plugins: All other major functionality within OpenSearch Dashboards.
+
+Plugins are a way to extend and customize the core functionality of OpenSearch Dashboards. They do not need to be a part of the Dashboards repository, though many are! They makeup some of the core applications and services within it.
+
+Plugins are classes that can be loaded via the [Dashboards plugin API](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1.2/src/core/CONVENTIONS.md#plugin-structure) to integrate with the core system via [lifecycle methods](#lifecycle-methods). Plugins can consist of either client side code (public), server side code (server), or both. Plugins can also interact with each other and core from both client and server. Plugins must also contain a manifest file that describes a set of properties, both required and optional that core system can use to load and initialize the plugin correctly.
+
+A plugin is usually made up of two parts:
+
+- Manifest file (`opensearch_dashboards.json`)
+- Plugin definition that implements an instance of the [Plugin](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1.2/src/core/public/plugins/plugin.ts#L45-L54) class
+
+If a plugin has a server and client side code, each section needs to describe the class separately. The interface for each remains the same. A typical plugin has the following folder structure:
+
+```sh
+my_plugin/
+├── opensearch_dashboards.json
+├── public
+│   ├── applications
+│   │   ├── my_app
+│   │   │   └── index.ts
+│   │   └── index.ts
+│   ├── services
+│   │   ├── my_service
+│   │   │   └── index.ts
+│   │   └── index.ts
+│   ├── index.ts
+│   └── plugin.ts
+└── server
+    ├── routes
+    │   └── index.ts
+    ├── collectors
+    │   └── register.ts
+    ├── saved_objects
+    │   ├── index.ts
+    │   └── my_type.ts
+    ├── services
+    │   ├── my_service
+    │   │   └── index.ts
+    │   └── index.ts
+    ├── index.ts
+    └── plugin.ts
+```
+
+More details on the plugin structure can be found in the [Conventions readme under the core](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1.2/src/core/CONVENTIONS.md#plugin-structure).
+
+### Manifest file
+
+The role of the manifest file is to describe the set of required and optional properties of a plugin such as plugin name version and other required plugins.
+
+The manifest file signature is defined by the interface [`PluginManifest`](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1.2/src/core/server/plugins/types.ts#L126-L196)
+
+e.g. manifest file
```

Comment:
drop this line

Reply:
The example or the link to the manifest file? also might i ask why? This is very useful to someone creating their own manifest file since we dont have docs for the manifest file right now. This is the best resource we have at the moment.

Reply:
Just to be sure, you're asking me to drop only line 76 right?